scripts/publish-preview.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { join } = require('path');
+const glob = require('glob').sync;
+const tiny = require('tiny-json-http');
+const fs = require('fs');
+const { gzipSync } = require('zlib');
+
+async function go() {
+	const { TRAVIS_PULL_REQUEST, TRAVIS_PULL_REQUEST_SHA } = process.env;
+
+	if (!TRAVIS_PULL_REQUEST) { throw new ReferenceError('Missing env var TRAVIS_PULL_REQUEST'); }
+	if (!TRAVIS_PULL_REQUEST_SHA) { throw new ReferenceError('Missing env var TRAVIS_PULL_REQUEST_SHA'); }
+
+	const dir = join(__dirname, '..', 'out');
+	const files = glob(join(dir, '**'), { nodir: true });
+
+	if (!files.length) { throw new ReferenceError('No preview files found to publish'); }
+
+	console.log(`Publishing preview build of PR ${TRAVIS_PULL_REQUEST} (SHA ${TRAVIS_PULL_REQUEST_SHA})`);
+
+	const data = {
+		pr: TRAVIS_PULL_REQUEST,
+		sha: TRAVIS_PULL_REQUEST_SHA,
+		files: [],
+	};
+	for (const file of files) {
+		const filename = file.replace(dir, '').slice(1);
+		const contents = fs.readFileSync(file);
+		const body = gzipSync(contents).toString('base64');
+		console.log(`Packaging: ${filename} (${body.length / 1000}KB)`);
+		data.files.push({
+			filename: filename,
+			body: body,
+		});
+	}
+
+	const url = 'https://ci.tc39.es/preview/tc39/ecma262';
+
+	const payloadSize = JSON.stringify(data).length;
+	console.log(`Payload size: ${payloadSize / 1000}KB`);
+	if (payloadSize >= 1000 * 1000 * 6) {
+		throw Error('Payloads must be under 6MB');
```

Comment:
Same here

Reply:
Changed the others to `kB`, although correct me if I'm wrong, but I do believe the proper notation is `MB`. This is backed up by the `Intl.NumberFormat`:
```js
const formatter = new Intl.NumberFormat('en', {
  style: 'unit',
  unit: 'megabyte',
});
formatter.format(1.234);
// → '1.234 MB'
```

Reply:
Yes, `MB` is correct, but it must be preceded by a space.

Reply:
Interesting, is there a spec for that? I’ve always wondered.

Reply:
Yes there is! http://cldr.unicode.org/ It's the data source the `Intl.*` specs use.

Reply:
Neat! Where specifically is that part of the spec published?

Reply:
Eh, the data is separate, but I guess you're asking about https://unicode.org/reports/tr35/tr35-numbers.html?

Reply:
Hmm, I’m not able to find anything there pertaining to use of space between numbers and units of measurement for data.

Reply:
The spacing and positioning rules are locale-specific, so they're encoded in the CLDR data itself.

Reply:
This wasn’t actually resolved

Reply:
It’s debugging info, primarily meant for me alone, since I’m the only one that’ll be maintaining it, that users never see. I consider it sufficiently resolved.